src/NzbDrone.Common/UniqueIdentifier/UniqueIdentifierGenerator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using NzbDrone.Common.Cache;
+
+namespace NzbDrone.Common.UniqueIdentifier
+{
+    public interface IUniqueIdentifierGenerator
+    {
+        int Get(string key);
+    }
+
+    public class UniqueIdentifierGenerator : IUniqueIdentifierGenerator
+    {
+        private static int _lastIdentifier;
+
+        private ICached<int> _cache;
+
+        public UniqueIdentifierGenerator(ICacheManager cacheManager)
+        {
+            _cache = cacheManager.GetCache<int>(typeof(UniqueIdentifierGenerator));
+        }
+
+        public int Get(string key)
+        {
+            return _cache.Get(key, GetNextIdentifier, TimeSpan.FromHours(24));
```

Comment:
I don't think we have a big enough keyspace. The Downloadclient id stuff is a full string, convert that to base10 and you'll have a very big number, not valid JSON integer.
The UniqueIdentifierGenerator is intended to guarantee there won't be collisions on the UI.